plugins/rest/auth.go

```diff
@@ -169,6 +174,95 @@ type tokenEndpointResponse struct {
 	ExpiresIn   int64  `json:"expires_in"`
 }
 
+type awsKmsKeyConfig struct {
+	Name      string `json:"name"`
+	Algorithm string `json:"algorithm"`
+}
+
+func converSignatureToBase64(alg string, der []byte) (string, error) {
+	r, s, derErr := pointsFromDER(der)
+	if derErr != nil {
+		return "", fmt.Errorf("failed to read points from der %v", derErr)
+	}
+
+	signatureData, err := convertPointsToBase64(alg, r.Bytes(), s.Bytes())
+	if err != nil {
+		return "", err
+	}
+	return signatureData, nil
+}
+
+func pointsFromDER(der []byte) (R, S *big.Int, err error) {
+	R, S = &big.Int{}, &big.Int{}
+	data := asn1.RawValue{}
+	if _, err := asn1.Unmarshal(der, &data); err != nil {
+		return nil, nil, fmt.Errorf("failed to unmarshall the signature from DER format %v", err)
+
+	}
+	// The format of our DER string is 0x02 + rlen + r + 0x02 + slen + s
```

Comment:
Is this based on some standard? If yes, please include the references.

Reply:
Some of the references are mentioned in the commit
References:
1) https://github.com/go-jose/go-jose/blob/v3/asymmetric.go#L501
2) https://github.com/codelittinc/gobitauth/blob/master/sign.go#L101

Reply:
Added some references